src/bundle.c

```diff
@@ -2502,6 +2502,50 @@ gboolean load_manifest_from_bundle(RaucBundle *bundle, RaucManifest **manifest,
 	return res;
 }
 
+static gboolean read_complete_dm_device(gchar *dev, GError **error)
+{
+	int fd = -1;
+	g_autofree void* buf = NULL;
+	ssize_t r;
+	gboolean ret = TRUE;
+
+	g_return_val_if_fail(dev != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	fd = g_open(dev, O_RDONLY | O_CLOEXEC, 0);
+	if (fd < 0) {
+		int err = errno;
+		g_set_error(error,
+				G_FILE_ERROR,
+				g_file_error_from_errno(err),
+				"Failed to open %s: %s", dev, g_strerror(err));
+		return FALSE;
+	}
+
+	lseek(fd, 0, SEEK_SET);
+
+	buf = g_malloc0(65536);
+
+	for (guint sector = 0;; sector++) {
```

Comment:
This should use `goffset` instead of `guint`, as it's operating on file offsets. Also it's not iterating over sectors (which are usually either 512 or 4Ki bytes on Linux), but 64kiB. Perhaps just call it `i` or `chunk`.